rfcs/chromium-connector.md

```diff
@@ -0,0 +1,213 @@
+| Response notification | [`page.on('response')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-response) |
+| Close browser         | [`browser.close()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browserclose) |
+| Navigate              | [`browser.newPage()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browsernewpage) <br> [`page.goto(url[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagegotourl-options) <br> [`page.waitForNavigation([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagewaitfornavigationoptions) |
+| Set page content | [`page.setContent(html[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagesetcontenthtml-options) |
+| Input | [`page.type(selector, text[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagetypeselector-text-options) <br> [`page.click(selector[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pageclickselector-options) |
+
+
+## Other comments/questions
+
+**How to run multiple pages in the same browser from different Puppeteer instances?**
+
+The idea is to use a similar approach where the `PID` is stored in a file and alternate between [`puppeteer.launch([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerlaunchoptions) and [`puppeteer.connect(options)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerconnectoptions) when the browser is already running (using `browserWSEndpoint`).
+
+**How to run a different Chromium based browser?**
+
+[`puppeteer.launch([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerlaunchoptions) has an option to pass `executablePath`. An option should be exposed in `connector-chromium` configuration to specify it. This value could be the `path` or maybe a well-known `string` value (`chromium-canary`, `chrome`, etc.) and someone (CLI, `connetor`?) could resolve the path similar to what `chrome-launcher` does today to decide what browser to run.
+
+Another thing to take into account is if the user starts a session with one `chromium` browser and decides to launch another session with another while it is still running. The `cdp.pid` (or whatever new file will be used) should probably contain information about the executable path to use or maybe this is an scenario that is not supported.
+
+**Package updates**
+
+* `chrome-connector` and `utils-debugging-protocol-common` should be deprecated and point users to download the new one.
+* `configuration-*` packages should be updated to use `connector-chromium`
+
+**Viewport**
+
+Puppeteer's viewport default to 800x600 viewport. `null` disables the default viewport. This should be exposed in the options somehow and I think we should default to `null` in our case.
```

Comment:
While I agree I'd personally more commonly want `null`, perhaps we should stick closely to `puppeteer`'s defaults, making it easier to just forward an entire group of options along without further processing by us. Thoughts?

Reply:
Not sure about sticking with puppeteer in this one, I think my watch has a higher resolution than that and 800x600... Don't know how representative that resolution is right now. Because we are running on the users machine I think we should use the whole real state and for the online scanner look into the most popular screen resolution and use that one.

Reply:
@webhintio/contributors thoughts about this? ☝

Reply:
FWIW `puppeteer` is planning to [raise the default size in 2.0](https://github.com/GoogleChrome/puppeteer/issues/2079):
> set initial viewport size to 1280 x 1024

That said, I'm okay with us defaulting to `null` and letting our users set an explicit size if they want one (e.g. for consistent results across test runs).

Reply:
Cool, I'm ok with reviewing the defaults once `puppeteer` 2 is out. The proposed value seems more reasonable 😊